api-tests/src/test/java/org/openmrs/module/reporting/cohort/definition/evaluator/DrugOrderCohortDefinitionEvaluatorTest.java

```diff
@@ -0,0 +1,185 @@
+import org.junit.Before;
+import org.junit.Ignore;
+import org.junit.Test;
+import org.openmrs.Cohort;
+import org.openmrs.Concept;
+import org.openmrs.api.context.Context;
+import org.openmrs.module.reporting.cohort.definition.DrugOrderCohortDefinition;
+import org.openmrs.module.reporting.cohort.definition.service.CohortDefinitionService;
+import org.openmrs.module.reporting.common.DateUtil;
+import org.openmrs.test.BaseModuleContextSensitiveTest;
+
+public class DrugOrderCohortDefinitionEvaluatorTest extends BaseModuleContextSensitiveTest {
+  protected static final String TEST_DATA = "org/openmrs/module/reporting/include/ReportTestDataset-openmrs-1.10.xml";
+  private DrugOrderCohortDefinition cohortDefinition;
+
+  @Before
+  public void setup() throws Exception {
+    cohortDefinition = new DrugOrderCohortDefinition();
+    executeDataSet(TEST_DATA);
+  }
+
+  @After
+  public void tearDown() {
+    cohortDefinition = null;
+  }
+  
+  @Test
+  public void evaluate_shouldReturnAllPatientsCurrentlyActiveOnDrugs() throws Exception {
+    cohortDefinition.setOnlyCurrentlyActive(true);
+    Cohort cohort = Context.getService(CohortDefinitionService.class).evaluate(cohortDefinition, null);
+    System.out.println("Currently on Drugs " + cohort.getMemberIds() + " " + cohortDefinition + "===");
+  }
+  
+  @Test
+  public void evaluate_shouldReturnAllPatientsCurrentlyNotActiveOnDrugs() throws Exception {
+    cohortDefinition.setOnlyCurrentlyActive(false);
+    Cohort cohort = Context.getService(CohortDefinitionService.class).evaluate(cohortDefinition, null);
+    System.out.println("Currently Not on Drugs " + cohort.getMemberIds() + " " + cohortDefinition + "===");
+  }
+  
+  @Test
+  public void evaluate_shouldReturnAllPatientsThatHaveTakenAnyofListedDrugs() throws Exception {
+    List drugSetList = new ArrayList<Concept>();
+    drugSetList.add(new Concept(88));
+    drugSetList.add(new Concept(792));
+    drugSetList.add(new Concept(20));
+    cohortDefinition.setDrugSets(drugSetList);
+    cohortDefinition.setWhich("ANY");
+    Cohort cohort = Context.getService(CohortDefinitionService.class).evaluate(cohortDefinition, null);
+    System.out.println("Taking Drugs Any" + cohort.getMemberIds()  + " " + cohortDefinition+ "===");
+    Assert.assertTrue(cohort.contains(2));
+    Assert.assertTrue(cohort.contains(21));
+    Assert.assertTrue(cohort.contains(22));
+    Assert.assertEquals(3, cohort.size());
+
+  }
+  
+  @Test
+  public void evaluate_shouldReturnAllPatientsThatHaveNeverTakenListedDrugs() throws Exception {
+    List drugSetList = new ArrayList<Concept>();
+    drugSetList.add(new Concept(88));
+    drugSetList.add(new Concept(792));
+    cohortDefinition.setDrugSets(drugSetList);
+    cohortDefinition.setWhich("NONE");
+    Cohort cohort = Context.getService(CohortDefinitionService.class).evaluate(cohortDefinition, null);
+    System.out.println("Taking Drugs None" + cohort.getMemberIds() +  " " + cohortDefinition + "===");
```

Comment:
Did you take a look at this? https://wiki.openmrs.org/display/docs/Java+Conventions#JavaConventions-Donotprintout

Reply:
removed